filecoindot-proofs/src/verify.rs

```diff
@@ -44,3 +30,21 @@ impl ProofVerify {
         Self::traverse_and_match::<N>(proof, index - 1, target_cid)
     }
 }
+
+impl Verify for ProofVerify {
+    /// Verify the proof and the the trie actually matches. Each cid in the proof
+    /// is connected to its neighbours. The proof should match exactly in path from
+    /// the root to the node.
+    /// Note that proof[proof.len-1] == root_cid_bytes. This function does not assume
+    /// the head of the proof to be equal to node_cid, as long as it's in the proof.
+    fn verify_proof<N>(proof: Vec<Vec<u8>>, node_cid: Vec<u8>) -> Result<(), Error>
+    where
+        N: GetCid + for<'de> serde::Deserialize<'de>,
+    {
+        let node_cid = Cid::try_from(node_cid).map_err(|_| Error::VerificationFailed)?;
```

Comment:
I think this should be a new variant `Error::CidError(cid::Error)` to just forward the cid parsing error.

Reply:
oh wait, nvm, this is a pallet error which doesn't need to be that detailed to save up memory space.